demo/fake-api.js

```diff
@@ -0,0 +1,614 @@
+            "template.name": "Demo Template",
+          },
+        },
+      });
+      websocket.send(
+        JSON.stringify({
+          meta: {
+            category: "ActiveRecordingCreated",
+            type: { type: "application", subType: "json" },
+            serverTime: +Date.now(),
+          },
+          message: {
+            target: request.params.targetId,
+            recording,
+          },
+        })
+      );
+      return recording;
+    });
+    this.get(
+      "/api/v1/targets/:targetId/recordings",
+      (schema) => schema.recordings.all().models
+    );
+    this.delete(
+      "/api/v1/targets/:targetId/recordings/:recordingName",
+      (schema, request) => {
+        let recordingName = request.params.recordingName;
+        let recording = schema.recordings.where({ name: recordingName });
+        schema.recordings.findBy({ name: recordingName }).destroy();
+        let msg = {
+          meta: {
+            category: "ActiveRecordingDeleted",
+            type: { type: "application", subType: "json" },
+            serverTime: +Date.now(),
+          },
+          message: {
+            recording: {
+              ...recording.models[0].attrs,
+            },
+            target: request.params.targetId,
+          },
+        };
+        websocket.send(JSON.stringify(msg));
+        return new Response(200);
+      }
+    );
+    this.patch(
+      "/api/v1/targets/:targetId/recordings/:recordingName",
+      (schema, request) => {
+        let body = request.requestBody;
+        let recordingName = request.params.recordingName;
+        let recording = schema.recordings.where({ name: recordingName });
+        let msg = {};
+        switch (body) {
+          case "STOP":
+            recording.update({ state: "STOPPED" });
+            msg = {
+              meta: {
+                category: "ActiveRecordingStopped",
+                type: { type: "application", subType: "json" },
+                serverTime: +Date.now(),
+              },
+              message: {
+                recording: {
+                  ...recording.models[0].attrs,
+                },
+                target: request.params.targetId,
+              },
+            };
+            websocket.send(JSON.stringify(msg));
+            break;
+          case "SAVE":
+            msg = {
+              meta: {
+                category: "ActiveRecordingSaved",
+                type: { type: "application", subType: "json" },
+                serverTime: +Date.now(),
+              },
+              message: {
+                recording: {
+                  ...recording.models[0].attrs,
+                  size: Math.ceil(Math.random() * 1000000),
+                  archivedTime: +Date.now(),
+                },
+                target: request.params.targetId,
+              },
+            };
+            websocket.send(JSON.stringify(msg));
+            schema.archives.create(recording.models[0].attrs);
```

Comment:
We need to specify the size of the recording as above too (in the notification message)?

![Screenshot from 2023-05-09 13-42-59](https://github.com/cryostatio/cryostatio.github.io/assets/68053619/9b426447-9606-4561-bcd9-9ddd5ca11416)

Reply:
@andrewazores Still need to update this?

Reply:
We might just copy the noti message into `create`?

```js
{
    ...recording.models[0].attrs,
    size: size,
    archivedTime: +Date.now(),
}
```